src/coreclr/src/gc/gcpriv.h

```diff
@@ -693,32 +694,62 @@ class alloc_list
 
 class allocator
 {
-    size_t num_buckets;
-    size_t frst_bucket_size;
+    int first_bucket_bits;
+    unsigned int num_buckets;
     alloc_list first_bucket;
     alloc_list* buckets;
     alloc_list& alloc_list_of (unsigned int bn);
     size_t& alloc_list_damage_count_of (unsigned int bn);
 
 public:
-    allocator (unsigned int num_b, size_t fbs, alloc_list* b);
+    allocator (unsigned int num_b, int fbb, alloc_list* b);
+
     allocator()
     {
         num_buckets = 1;
-        frst_bucket_size = SIZE_T_MAX;
+        first_bucket_bits = sizeof(size_t) * 8;
```

Comment:
`first_bucket_bits` is a compile-time constant; might get slightly better codegen with something like:

```c
private:
    constexpr size_t first_bucket_bits() { return sizeof(size_t) * 8; }
```

Reply:
Ah, nevermind. Missed the other constructor with a `fbb` parameter.

Reply:
There are two constructors. Another takes parameters.